mobile/src/main/java/org/openhab/habdroid/ui/WidgetAdapter.kt

```diff
@@ -962,73 +970,107 @@ class WidgetAdapter(
         }
 
         override fun onStart() {
-            if (mediaPlayer.currentMediaItem != null) {
-                mediaPlayer.play()
+            if (exoPlayer.playbackState != Player.STATE_IDLE) {
+                exoPlayer.playWhenReady = true
             }
         }
 
         override fun onStop() {
-            if (mediaPlayer.currentMediaItem != null) {
-                mediaPlayer.pause()
+            if (exoPlayer.playbackState != Player.STATE_IDLE) {
+                exoPlayer.playWhenReady = false
             }
         }
 
         private fun loadVideo(widget: Widget, forceReload: Boolean) {
+            playerView.isVisible = true
             errorView.isVisible = false
             loadingIndicator.isVisible = true
-            val mediaItem = determineVideoUrlForWidget(widget)?.let { url ->
-                val meta = MediaMetadata.Builder().putString(MediaMetadata.METADATA_KEY_TITLE, widget.label).build()
-                UriMediaItem.Builder(url.toUri())
-                    .setMetadata(meta)
-                    .build()
+
+            SSLContext.getInstance("TLS").apply {
```

Comment:
This is required to connect to a server that has an untrusted certificate, but I'm not sure if the hostname verifier introduces any security issues.